core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -784,6 +784,11 @@ public boolean supportsWindowFunctions() {
     return true;
   }
 
+  /** Returns whether this dialect supports case when return boolean type. */
+  public boolean supportBooleanCaseWhen() {
```

Comment:
does the old Oracle dialect support Booleans in other expressions?

Reply:
What I'm trying to convey is that CASE WHEN statements are used as predicates in filters or concatenation conditions and return a value of bool.  could you make some suggestions for better naming of the method? 
edit: ... sry, I repeated the edit multiple times

Reply:
Is this about CASE expressions or about the BOOLEAN data type not being supported?
Or is it about CASE expressions that evaluate to BOOLEAN results not being supported - which is what you imply?

Reply:
It is the latter，about CASE expressions that evaluate to BOOLEAN results not being supported

Reply:
In that case the name of the function is adequate.
Just to make sure: you say that an expression such as "TRUE" would work in a WHERE expression?

Reply:
Did you say "true" literal?， I have supported in [CALCITE-6482](https://github.com/apache/calcite/pull/3863)

Reply:
can you add a test that combines the two, e.g., `CASE WHEN x > 1 THEN TRUE ...`?